typed-racket-doc/typed-racket/scribblings/guide/occurrence.scrbl

```diff
@@ -84,6 +86,51 @@ the typechecker learns from the result of applying the function:
 Predicates for all built-in types are annotated with similar propositions
 that allow the type system to reason logically about predicate checks.
 
+@subsection{Specifying Propositions}
+
+While propositions are provided for all built-in type predicates,
+we may want to provide propositions for our own predicates as well.
+For instance, consider the following predicate,
+which determines whether a given list contains only strings.
+Intuitively, a value that satisfies the predicate must have type
+@racket[(Listof String)].
+
+@examples[#:no-result #:eval the-eval*
+  (: listof-string? (-> (Listof Any) Boolean))
+  (define (listof-string? lst)
+    (andmap string? lst))
+]
+
+We then may wish to use this predicate to narrow a type in our main program:
+
+@examples[#:label #f #:eval the-eval*
+  (: main (-> (Listof Any) String))
+  (eval:error (define (main lst)
+                (cond
+                  [(listof-string? lst) (first lst)]
```

Comment:
Perhaps we need to mention `(andmap string? lst)` works here. After all, `listof-string` is just an eta-expansion of (curried) `(andmap string?). 

Also, on a possibly related note,  the reason `(: listof-string? (-> (Listof Any) Boolean))` does not work is that its positive and negative propositions are `always-true`, which is bigger than `lst is (Listof String)` or `lst is not (Listof String)`

Reply:
To the first point, we could point out that the typechecker can infer proposition types for expressions that return booleans (in the untyped Racket sense).

Reply:
@capfredf does this look better? https://github.com/micahcantor/typed-racket/pull/2.

I actually kinda like the current version. I definitely don't want to get to how `(-> (Listof Any) Boolean)` technically has a proposition, since it's likely confusing to beginners who read it (keep in mind that this is the Typed Racket Guide). 

The narrative in the above PR works, but it kinda takes too long to set things up before getting to the point. Though that may be fine after all.

Reply:
I think I prefer the current version too, looking at it side by side. Personally, the issue I had lied more in how to annotate the type of a proposition at the top level, and less with understanding that occurrence typing worked on more complex predicates.

Reply:
I think the current version looks good. Just in this particular example, `(andmap string? lst`) works directly. Perhaps, we can simply add something like `if we want to turn `(andmap string? x)` into a custom predicate, here is what we need to do:`

Reply:
I'll work on a margin note as a separate commit to address @capfredf's concern.